appmetrics/app_metrics.go

```diff
@@ -92,15 +92,17 @@ func (am *AppMetrics) updateCacheLoop() {
 			// update the database after closing the app map
 			// since this won't affect the app map, no need to continue touching it
 			am.db.Batch(func(tx *bolt.Tx) error {
+				b, err := tx.CreateBucketIfNotExists(am.appBucket)
```

Comment:
Why is this global now?

Reply:
the update of app was previously done inside the delete loop, which is wrong, they are two separate operations. I put the update outside, and so I need to declare this variable outside as well.